### PR TITLE
[P2T support No.51-55] standard_gamma  slice_scatter  standard_normal stanh strided_slice

### PR DESCRIPTION
**Pull request by mzj104**
修改内容 **（以新rule修改）**

------------ 
这5个api都需要编写rules

> slice_scatte 无对应api
> strided_slice 无对应api
> standard_gamma 对应 torch.distributions.Gamma ， 但需要预处理和后处理
> standard_normal 对应 torch.randn，但需要预处理
> stanh 对应 torch.tanh， 但需要预处理和后处理

----------------------

standard_normal   和  standard_gamma 是随机api，所以放到了tester/api_config/api_config_stochastic.txt 中

----------------------
standard_gamma 在 初始化的时候存在问题，修改了初始化的代码， 保证为正数：
```
            elif api_config.api_name == "paddle.standard_gamma":
                self.numpy_tensor =numpy.random.random(self.shape).astype(self.dtype)
```

- https://github.com/PaddlePaddle/Paddle/issues/72643

### CODE REVIEW

File: tester/base.py [hunk 1]
```diff
@@ -32,8 +32,6 @@
     "paddle.randint",
     "paddle.randint_like",
     "paddle.randperm",
-    "paddle.standard_gamma",
-    "paddle.standard_normal",
```
Comment (cangtianhuang) [inline tester/base.py:36]: 随机 api 我们需要支持转换，但是不需要进行精度对比，此处的列表中需要保留，并将 standard_normal 和 standard_gamma 的所有配置移入 tester/api_config/api_config_stochastic.txt 中

File: tester/accuracy.py [hunk 2]
```diff
@@ -185,6 +185,8 @@ def test(self):
         torch.cuda.empty_cache()
 
         try:
+            if 'standard_gamma' or 'standard_normal' in self.api_config.api_name:
+                paddle.seed(42)
```
Comment (cangtianhuang) [inline tester/accuracy.py:189]: 不必设置种子，可以删去

File: tester/paddle_to_torch/rules.py [hunk 3]
```diff
@@ -2558,6 +2558,97 @@ def apply(self, paddle_api: str) -> ConvertResult:
...
+shape = list(x.shape)
+for i, axis in enumerate(axes):
+    start, end, stride = starts[i], ends[i], strides[i]
+    shape[axis] = (end - start + stride - 1) // stride
+if list(value.shape) != shape:
+    value = value.expand(shape)
+result = x.clone()
+result[tuple(slices)] = value
+"""
+        code = Code(core=[core])
```
Comment (cangtianhuang) [inline tester/paddle_to_torch/rules.py:2576]:
```suggestion
        code = Code(core=core.splitlines())
```

File: tester/paddle_to_torch/rules.py [hunk 4]
```diff
@@ -2558,6 +2558,97 @@ def apply(self, paddle_api: str) -> ConvertResult:
...
+        return ConvertResult.success(paddle_api, code, is_torch_corresponding=False)
+
+class StandardGammaRule(BaseRule):
+    def apply(self, paddle_api: str) -> ConvertResult:
+        defaults_code, map_code = self.apply_generic()
+        pre = """
+rate = torch.ones_like(x)
+torch.manual_seed(42)
```
Comment (cangtianhuang) [inline tester/paddle_to_torch/rules.py:2584]: 若仅支持转换的话，此处不必设置种子

File: tester/paddle_to_torch/rules.py [hunk 5]
```diff
@@ -2558,6 +2558,97 @@ def apply(self, paddle_api: str) -> ConvertResult:
...
+"""
+        core = f"result = {self.torch_api}(**_kwargs)"
+        post = """
+result = result.sample()
+"""
+        code = Code(
+            preprocess=defaults_code + pre.splitlines() + map_code,
+            core=core.splitlines(),
+            postprocess=post.splitlines(),
+        )
```
Comment (cangtianhuang) [inline tester/paddle_to_torch/rules.py:2594]: 可改为（不影响运行）
```suggestion
        core = f"result = {self.torch_api}(**_kwargs)"
        post = "result = result.sample()"
        code = Code(
            preprocess=defaults_code + pre.splitlines() + map_code,
            core=[core],
            postprocess=[post],
        )
```

File: tester/paddle_to_torch/rules.py [hunk 6]
```diff
@@ -2558,6 +2558,97 @@ def apply(self, paddle_api: str) -> ConvertResult:
...
+        return ConvertResult.success(paddle_api, code)    
+
+class StandardNormalRule(BaseRule):
+    def apply(self, paddle_api: str) -> ConvertResult:
+        defaults_code, map_code = self.apply_generic()
+        pre = """
+torch.manual_seed(42)
```
Comment (cangtianhuang) [inline tester/paddle_to_torch/rules.py:2601]: 同理

File: tester/paddle_to_torch/rules.py [hunk 7]
```diff
@@ -2558,6 +2558,97 @@ def apply(self, paddle_api: str) -> ConvertResult:
...
+"""
+        core = f"result = {self.torch_api}(**_kwargs)"
+        post = """
+result = result * scale_b
+"""
+        code = Code(
+            preprocess=defaults_code + pre.splitlines() + map_code,
+            core=core.splitlines(),
+            postprocess=post.splitlines(),
+        )
```
Comment (cangtianhuang) [inline tester/paddle_to_torch/rules.py:2624]: 可修改为（可不修改）
```suggestion
        pre = "x = x * scale_a"
        core = f"result = {self.torch_api}(**_kwargs)"
        post = "result = result * scale_b"
        code = Code(
            preprocess=defaults_code + [pre] + map_code,
            core=[core],
            postprocess=[post],
        )
```

**Review by cangtianhuang — CHANGES_REQUESTED** [review]
随机 api 的位置请改一下，不必设置随机种子

**Review by cangtianhuang — APPROVED** [review]
LGTM

**Review by wanghuancoder — APPROVED** [review]
LGTM
